examples/apps/tree-comparison/src/model/legacyTreeInventoryList.ts

```diff
@@ -0,0 +1,244 @@
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const inventoryNodeId = rootNode.traits.get("inventory" as TraitLabel)![0];
+		this.tree.applyEdit(
+			Change.insertTree(
+				addedNode,
+				StablePlace.atEndOf({
+					parent: inventoryNodeId,
+					label: "inventoryItems" as TraitLabel,
+				}),
+			),
+		);
+	};
+
+	public readonly getItems = (): IInventoryItem[] => {
+		return [...this._inventoryItems.values()];
+	};
+
+	protected async initializingFirstTime(): Promise<void> {
+		const legacySharedTree = this.runtime.createChannel(
+			undefined,
+			LegacySharedTree.getFactory().type,
+		) as LegacySharedTree;
+
+		const inventoryNode: BuildNode = {
+			definition: "inventory",
+			traits: {
+				// REV: If I try to remove these sample items (leaving inventoryItems as an empty array)
+				// the trait is removed entirely (undefined).  It seems that empty traits are discarded?
+				// This is a problem when later I want to iterate over it (in hasInitialized, when
+				// building up my initial set of InventoryItem objects), since I'll get a not-iterable error.
+				inventoryItems: [
+					{
+						definition: "inventoryItem",
+						traits: {
+							// REV: I tried adding an explicit "ID" trait here and using that for tracking,
+							// rather than the NodeId itself.  However, when deleting a node the "removed" member
+							// of the "viewChange" event args only includes the NodeId and the nodes are already
+							// gone (cannot be retrieved).  So the ID is lost at that point.  I could maintain a
+							// separate NodeId -> ID mapping to track, but this feels kind of clunky and defeats
+							// the point of getting away from direct usage of NodeIds.
+							name: {
+								definition: "name",
+								payload: "nut",
+							},
+							quantity: {
+								definition: "quantity",
+								payload: 0,
+							},
+						},
+					},
+					{
+						definition: "inventoryItem",
+						traits: {
+							name: {
+								definition: "name",
+								payload: "bolt",
+							},
+							quantity: {
+								definition: "quantity",
+								payload: 0,
+							},
+						},
+					},
+				],
+			},
+		};
+		legacySharedTree.applyEdit(
+			Change.insertTree(
+				inventoryNode,
+				StablePlace.atStartOf({
+					parent: legacySharedTree.currentView.root,
+					label: "inventory" as TraitLabel,
+				}),
+			),
+		);
+
+		this.root.set(legacySharedTreeKey, legacySharedTree.handle);
+	}
+
+	/**
+	 * hasInitialized is run by each client as they load the DataObject.  Here we use it to set up usage of the
+	 * DataObject, by registering an event listener for changes to the inventory list.
+	 */
+	protected async hasInitialized() {
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		this._tree = await this.root
+			.get<IFluidHandle<LegacySharedTree>>(legacySharedTreeKey)!
+			.get();
+
+		// REV: Not exactly sure why a checkout is required to get "viewChange" events, seems like it should be able
```

Comment:
Comes down to layering. In the end what checkout gives you architecturally is a bit half-baked in legacy shared-tree. At some point there were goals toward a more comprehensive branching story, and checkout meant to fit into that puzzle, but we never ended up investing in it too fully.

Reply:
Seems not ideal, but I'm guessing we're not making further changes to legacy shared tree.  I converted this to just a normal comment explaining why we're using a checkout in latest revision.